src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -280,7 +286,15 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
             services.AddSingleton<IMetricsPublisher>(s =>
             {
                 var environment = s.GetService<IEnvironment>();
-                if (environment.IsFlexConsumptionSku())
+                if (environment.IsV1LinuxConsumptionOnLegion())
+                {
+                    var logger = s.GetService<ILogger<LinuxContainerLegionMetricsPublisher>>();
+                    var metricsTracker = s.GetService<ILinuxConsumptionMetricsTracker>();
+                    var standbyOptions = s.GetService<IOptionsMonitor<StandbyOptions>>();
+                    //var metricsLogger = s.GetService<IMetricsLogger>();
```

Comment:
TODO: Once we fix the cyclic ref update this code

Reply:
Done